energy-integration/farm-boosted-yields/src/lib.rs

```diff
@@ -60,29 +65,14 @@ pub trait FarmBoostedYieldsModule:
         SplitReward::new(base_farm_amount, boosted_yields_cut)
     }
 
-    fn collect_rewards(
-        &self,
-        week: Week,
-        reward_token_id: &TokenIdentifier,
-    ) -> TokenAmountPairsVec<Self::Api> {
-        let rewards_mapper = self.accumulated_rewards_for_week(week);
-        let total_rewards = rewards_mapper.get();
-        rewards_mapper.clear();
-
-        ManagedVec::from_single_item(TokenAmountPair::new(reward_token_id.clone(), total_rewards))
-    }
-
     fn claim_boosted_yields_rewards(
         &self,
         user: &ManagedAddress,
         farm_token_nonce: Nonce,
-        reward_token_id: &TokenIdentifier,
+        _reward_token_id: &TokenIdentifier,
     ) -> BigUint {
-        let rewards = self.claim_multi(
-            user,
-            farm_token_nonce,
-            |sc_ref: &Self, week: Week| Self::collect_rewards(sc_ref, week, reward_token_id),
-        );
+        let wrapper = FarmBoostedYieldsWrapper::new(farm_token_nonce);
+        let rewards = self.claim_multi::<FarmBoostedYieldsWrapper<Self>>(&wrapper, user);
```

Comment:
Type should be deduce automatically, I believe.
```suggestion
        let rewards = self.claim_multi(&wrapper, user);
```

Reply:
Done.